src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const count in actions) {
+    switch (actions[count].type) {
+      case 'addProperties':
+        for (const key in actions[count].extraData) {
+          state[key] = actions[count].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key in actions[count].keysToRemove) {
```

Comment:
This one can be done easier, you can loop over destructured keysToRemove and delete state properties with assigned keys.